File: Matlab/GeomagneticField/schLegendre.m
function p = schLegendre(n, m, x)
% Calculate Schmidt seminormalized associated Legendre polynomial

if m == 0
    p = legendrepoly(n, x);
elseif m > 0
    p = (-1)^m * sqrt(2 * factorial(n - m) / factorial(n + m)) * assLegendre(n,m,x);
else
    error('Order m cannot be negative')
end

File: source/geomagneticfield/sch_legendre.c
double SchLegendre(double n, double m, double x) {
  double p = 0;
  if (m == 0)
    p = Legendrepoly(n, x);
  else if (m > 0)
    p = pow((-1), m) * sqrt(2 * factorial(n-m)/factorial(n+m)) * AssLegendre(n, m, x);
  return p;
}
